paas_app_charmer/charm_state.py

```diff
@@ -205,20 +208,23 @@ class IntegrationsState:
         databases_uris: Map from interface_name to the database uri.
         s3_parameters: S3 parameters.
         saml_parameters: SAML parameters.
+        rabbitmq_parameters: RabbitMQ parameters.
     """
 
     redis_uri: str | None = None
     databases_uris: dict[str, str] = field(default_factory=dict)
     s3_parameters: "S3Parameters | None" = None
     saml_parameters: "SamlParameters | None" = None
+    rabbitmq_parameters: "RabbitMQParameters | None" = None
 
     @classmethod
-    def build(
+    def build(  # pylint: disable=too-many-arguments
```

Comment:
Isn't it a sign that we should split the charm_state is smaller data structure used within a smaller scope ?

Reply:
I think so as well, at least there should be a note explaining the disable

Reply:
This is the build function in IntegrationsState, that currently is something like:
```
    redis_uri: str | None = None
    databases_uris: dict[str, str] = field(default_factory=dict)
    s3_parameters: "S3Parameters | None" = None
    saml_parameters: "SamlParameters | None" = None
    rabbitmq_uri: str | None = None
```
I do not think we should split those fields apart yet. The function `build` has an extra parameter (cls) and for that reason there are 6 paramaters and the linter complains.

However, I agree that we have several fields "percolating" from the `CharmState.from_charm`. I had a try in https://github.com/canonical/paas-app-charmer/pull/41 creating the IntegrationsState outside of this package. 

Maybe we could discuss how to refactor the integrations and tackle this possible refactor in another PR, WDYT?

Reply:
Ok, please make a story it